arangod/IResearch/IResearchFilterFactory.cpp

```diff
@@ -908,10 +928,10 @@ Result fromGeoDistanceInterval(irs::boolean_filter* filter,
       return {TRI_ERROR_INTERNAL, "Malformed search/filter context"};
     }
 
-    auto& geo_filter =
-        (aql::NODE_TYPE_OPERATOR_BINARY_NE == node.cmp
-             ? filter->add<irs::Not>().filter<GeoDistanceFilter>()
-             : filter->add<GeoDistanceFilter>());
+    auto& geo_filter = (aql::NODE_TYPE_OPERATOR_BINARY_NE == node.cmp
+                            ? append<irs::Not>(*filter, ctx.hasNestedFields)
+                                  .filter<GeoDistanceFilter>()
```

Comment:
Should we always use `append` for filter combining? As technically  any filter could use `all` inside